ext/intl/rangeformatter/rangeformatter_class.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Bogdan Ungureanu <bogdanungureanu21@gmail.com>              |
+   +----------------------------------------------------------------------+
+*/
+
+#include <unicode/numberrangeformatter.h>
+#include <unicode/unumberrangeformatter.h>
+#include <unicode/numberformatter.h>
+#include <unicode/unistr.h>
+#include "../intl_convertcpp.h"
+
+extern "C" {
+    #include "php.h"
+    #include "../php_intl.h"
+    #include "../intl_data.h"
+    #include "rangeformatter_arginfo.h"
+    #include "rangeformatter_class.h"
+#include "intl_convert.h"
+}
+
+using icu::number::NumberRangeFormatter;
+using icu::number::NumberFormatter;
+using icu::number::UnlocalizedNumberFormatter;
+using icu::number::LocalizedNumberRangeFormatter;
+using icu::UnicodeString;
+using icu::MeasureUnit;
+
+static zend_object_handlers rangeformatter_handlers;
+zend_class_entry *class_entry_IntlNumberRangeFormatter;
+
+zend_object *IntlNumberRangeFormatter_object_create(zend_class_entry *ce)
+{
+	IntlNumberRangeFormatter_object* intern;
+
+	intern = (IntlNumberRangeFormatter_object*)zend_object_alloc(sizeof(IntlNumberRangeFormatter_object), ce);
+	zend_object_std_init(&intern->zo, ce);
+	object_properties_init(&intern->zo, ce);
+
+	return &intern->zo;
+}
+
+U_CFUNC PHP_METHOD(IntlNumberRangeFormatter, __construct)
+{
+   
```

Comment:
I think you could throw an exception if this functions is tried to be called: https://github.com/php/php-src/blob/8526de84a504cc3fead9a1734ea663e8fd2669f3/ext/dom/node.c#L2611